background/get-userscripts.js

```diff
@@ -17,61 +17,104 @@ function getL10NURLs() {
   return urls;
 }
 
+scratchAddons.localEvents.addEventListener("addonEnabled", ({ detail }) => {
+  const { addonId, manifest } = detail;
+  chrome.tabs.query({}, (tabs) =>
+    tabs.forEach((tab) => {
+      if (tab.url || (!tab.url && typeof browser !== "undefined")) {
+        chrome.tabs.sendMessage(tab.id, "getInitialUrl", { frameId: 0 }, async (res) => {
+          if (res) {
+            const { userscripts, userstyles } = await getAddonData({ addonId, url: res, manifest });
+
+            console.log(userscripts, userstyles);
+            chrome.tabs.sendMessage(
+              tab.id,
+              {
+                dynamicAddonEnabled: {
+                  scripts: userscripts,
+                  userstyles,
+                  addonId,
+                  injectAsStyleElt: manifest.injectAsStyleElt,
+                  index: scratchAddons.manifests.findIndex((addon) => addon.addonId === addonId),
+                },
+              },
+              { frameId: 0 }
+            );
+          }
+        });
+      }
+    })
+  );
+});
+scratchAddons.localEvents.addEventListener("addonDisable", ({ detail }) => {
+  const { addonId, manifest } = detail;
+  chrome.tabs.query({}, (tabs) =>
+    tabs.forEach((tab) => {
+      if (tab.url || (!tab.url && typeof browser !== "undefined")) {
+        chrome.tabs.sendMessage(tab.id, "getInitialUrl", { frameId: 0 }, async (res) => {
+          if (res) {
+            chrome.tabs.sendMessage(tab.id, { dynamicAddonDisable: { addonId } }, { frameId: 0 });
+          }
+        });
+      }
+    })
+  );
+});
+
+async function getAddonData({ addonId, manifest, url }) {
+  const promices = [];
+
+  const userscripts = [];
+  for (const script of manifest.userscripts || []) {
+    if (userscriptMatches({ url }, script, addonId))
+      userscripts.push({
+        url: script.url,
+        runAtComplete: typeof script.runAtComplete === "boolean" ? script.runAtComplete : true,
+      });
+  }
+  const userstyles = [];
+  for (const style of manifest.userstyles || []) {
+    if (userscriptMatches({ url }, style, addonId))
+      if (manifest.injectAsStyleElt) {
+        promices.push(
+          fetch(chrome.runtime.getURL(`/addons/${addonId}/${style.url}`))
+            .then((res) => res.text())
+            .then((text) => {
+              userstyles.push(text);
+            })
+        );
+      } else {
+        userstyles.push(chrome.runtime.getURL(`/addons/${addonId}/${style.url}`));
+      }
+  }
+  await Promise.all(promices);
+
+  return { userscripts, userstyles, promices };
+}
+
 async function getContentScriptInfo(url) {
   const data = {
     url,
     l10njson: getL10NURLs(),
     globalState: {},
     addonsWithUserscripts: [],
     addonsWithUserstyles: [],
-    themes: [],
   };
-  const fetchThemeStylesPromises = [];
-
-  for (const { addonId, manifest } of scratchAddons.manifests) {
+  for (let i = 0; i < scratchAddons.manifests.length; i++) {
```

Comment:
for of / for in should work
or even forEach!

Reply:
no i need a `for`, for a reason...
1. I need an index
2. iirc, I don't want to return or something, i forgot